### PR TITLE
:memo:(all)! improve the overall documentation

### LINKED ISSUE #158: :books: Add Documentation for OOD Metrics Computation and Interpretation
I would like to add a `METRICS.md` to domument the metrics used in the routines that come with the library.

For starters, I would like to document the OOD metrics used in `ClassificationRoutine`. As a first time user, I found it hard to interpret them without knowing the under-the-hood imlpementation details (#139). I reviewed the implementation of `ClassificationRoutine` and the `torchmetrics` binary classification metrics used to evaluate OOD performance. Could you please review the summary so that I can draft a document with more comprehensive doumentaiton?

1. **Input to OOD Metrics**:
   - The `ood_scores` tensor is passed as the `preds` tensor to the binary classification metrics.
   - The `ood_scores` are computed based on the `ood_criterion`:
     - `"msp"`: Negative maximum softmax probability (`-confs`).
     - `"logit"`: Negative maximum logit (`-logits.mean(dim=1).max(dim=-1).values`).
     - `"energy"`: Negative log-sum-exp of logits (`-logits.mean(dim=1).logsumexp(dim=-1)`).
     - `"entropy"`: Mean entropy of the predictions (`torch.special.entr(probs_per_est).sum(dim=-1).mean(dim=1)`).
     - `"mi"`: Mutual information of the ensemble predictions.
     - `"vr"`: Variation ratio of the ensemble predictions.
   - For `"msp"`, the `ood_scores` are in the range `[-1, -1/C]` for a C-class problem.

2. **Normalization of `ood_scores`**:
   - Since the binary classification metrics expect `preds` in the range `[0, 1]`, any values outside this range are interpreted as logits and passed through the sigmoid function.
   - For `ood_scores = [-1, -1/C]`, the sigmoid maps them to the range `[sig(-1), sig(-1/C)]`, which is approximately `(0.269, 0.5)` as C → ∞.

3. **Thresholding and Metric Computation**:
   - The binary classification metrics (e.g., AUROC, AUPR) dynamically compute thresholds based on the unique values in the `preds` tensor.
   - Any value above a threshold is treated as a positive prediction (indicating an OOD sample).
   - Lowering the threshold corresponds to lowering the confidence level at which a prediction is accepted as in-distribution (ID) vs. rejected as OOD.

4. **Interpretation**:
The metrics evaluate how well the model separates ID samples from OOD samples based on the confidence level. This assumes that the model uses a fixed "rejection" level during deployment. If the confidence of a prediction is below this level, it is rejected as OOD.

---

## Proposal

1. **Add a `METRICS.md` File**:
   - Include detailed documentation on how the OOD metrics are computed and how they should be interpreted.
   - Provide examples for each `ood_criterion` (e.g., `"msp"`, `"logit"`, `"energy"`, etc.) and explain how the `ood_scores` are derived.

2. **Clarify Metric Behavior in Code**:
   - Add# Add Documentation for OOD Metrics Computation and Interpretation

---

## My Understanding of OOD Metrics Computation

1. **Input to OOD Metrics**:
   - The `ood_scores` tensor is passed as the `preds` tensor to the binary classification metrics.
   - The `ood_scores` are computed based on the `ood_criterion`:
     - `"msp"`: Negative maximum softmax probability (`-confs`).
     - `"logit"`: Negative maximum logit (`-logits.mean(dim=1).max(dim=-1).values`).
     - `"energy"`: Negative log-sum-exp of logits (`-logits.mean(dim=1).logsumexp(dim=-1)`).
     - `"entropy"`: Mean entropy of the predictions (`torch.special.entr(probs_per_est).sum(dim=-1).mean(dim=1)`).
     - `"mi"`: Mutual information of the ensemble predictions.
     - `"vr"`: Variation ratio of the ensemble predictions.
   - For `"msp"`, the `ood_scores` are in the range `[-1, -1/C]` for a C-class problem.

2. **Normalization of `ood_scores`**:
   - Since the binary classification metrics expect `preds` in the range `[0, 1]`, any values outside this range are interpreted as logits and passed through the sigmoid function.
   - For `ood_scores = [-1, -1/C]`, the sigmoid maps them to the range `[sig(-1), sig(-1/C)]`, which is approximately `(0.269, 0.5)` as `C → ∞`.

3. **Thresholding and Metric Computation**:
   - The binary classification metrics (e.g., AUROC, AUPR) dynamically compute thresholds based on the unique values in the `preds` tensor.
   - Any value above a threshold is treated as a positive prediction (indicating an OOD sample).
   - Lowering the threshold corresponds to lowering the confidence level at which a prediction is accepted as in-distribution (ID) vs. rejected as OOD.

4. **Interpretation**:
The metrics evaluate how well the model separates ID samples from OOD samples based on the confidence level. This assumes that the model uses a fixed "rejection" level during deployment. If the confidence of a prediction is below this level, it is rejected as OOD.

---

## Proposal

1. **Add a `METRICS.md` File**:
   - Include detailed documentation on how the OOD metrics are computed and how they should be interpreted.
   - Provide examples for each `ood_criterion` (e.g., `"msp"`, `"logit"`, `"energy"`, etc.) and explain how the `ood_scores` are derived.

2. **Clarify Metric Behavior in Code**:
   - Add

### PR DESCRIPTION
**Pull request by o-laurent**
Specifically added a new page to explain the metrics & mathematical formulations for most of them. 

Close #158.

### CODE REVIEW

File: src/torch_uncertainty/ood_criteria.py [hunk 1]
```diff
@@ -58,10 +58,15 @@ class MaxLogitCriterion(TUOODCriterion):
...
-        """OOD criterion based on the maximum logit value.
+        r"""OOD criterion based on the Max-Logit score (Hendrycks et al.).
 
-        This criterion computes the negative of the highest logit value across
-        the output dimensions. Lower maximum logits indicate greater uncertainty.
+        Defined as the negative of the highest logit value (averaged over estimators
+        in the ensemble case):
+
+        .. math::
+            \text{score}(\mathbf{z}) = -\max_{i} z_i.
```
[the remark this answers is not in the document]
Comment (o-laurent) [reply src/torch_uncertainty/ood_criteria.py:67]: This is correct and should be fixed.
